analytics-core/src/main/java/com/liferay/analytics/client/android/Analytics.kt

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.analytics.client.android
+
+import android.annotation.SuppressLint
+import android.app.Application
+import android.content.Context
+import com.liferay.analytics.client.android.api.impl.AnalyticsClientImpl
+import com.liferay.analytics.model.AnalyticsEventsMessage
+import io.reactivex.Flowable
+import io.reactivex.annotations.NonNull
+import io.reactivex.annotations.Nullable
+import io.reactivex.schedulers.Schedulers
+
+/**
+ * @author Igor Matos
+ */
+class Analytics private constructor(val application: Application, private val analyticsKey: String,
+	private val flushIntervalInMilliseconds: Long) {
+
+	fun send(@NonNull eventId: String, @NonNull properties: Map<String, String>, @NonNull applicationId: String) {
+
+		if (userId == null) {
+			return
+		}
+
+		val analyticsEventsMessageBuilder = AnalyticsEventsMessage.builder(analyticsKey, userId)
+
+		val eventBuilder = AnalyticsEventsMessage.Event.builder(applicationId, eventId)
+
+		eventBuilder.properties(properties)
+
+		analyticsEventsMessageBuilder.event(eventBuilder.build())
+
+		var flowable = Flowable.fromCallable({
```

Comment:
You need backpressure?

Reply:
Yes, we need. But this is another task to do, so in the first moment this is the way we will send the events.

Reply:
Ok!